sources/packages/backend/apps/api/src/services/announcements/announcements.service.ts

```diff
@@ -0,0 +1,28 @@
+import { Injectable } from "@nestjs/common";
+import { Announcements, RecordDataModelService } from "@sims/sims-db";
+import { DataSource } from "typeorm";
+
+@Injectable()
+export class AnnouncementsService extends RecordDataModelService<Announcements> {
+  constructor(dataSource: DataSource) {
+    super(dataSource.getRepository(Announcements));
+  }
+
+  /**
+   * Get system announcements.
+   * @returns system announcements list.
+   */
+  async getAnnouncements(): Promise<Announcements[]> {
+    return this.repo.find({
+      select: {
+        id: true,
+        messageTitle: true,
+        message: true,
+        target: true,
+        startDate: true,
+        endDate: true,
+      },
+      order: { startDate: "ASC" },
```

Comment:
Are we not filtering the announcements based on the start and end dates here?

Reply:
updated